src/workflows-app/SubmitWorkflow.js

```diff
@@ -86,48 +94,79 @@ export const SubmitWorkflow = wrapWorkflowsPage({ name: 'SubmitWorkflow' })(
       load();
     });
 
+    useEffect(() => {
+      const refresh = async () => {
+        await refreshApps(true);
+      };
+
+      if (!currentApp || getIsAppBusy(currentApp)) {
+        refresh();
+      }
+    }, [currentApp, refreshApps]);
+
+    const createWorkflowsApp = Utils.withBusyState(setCreating, async () => {
+      try {
+        setCreating(true);
+        await Ajax(signal).Apps.createAppV2(Utils.generateAppName(), workspace.workspace.workspaceId, appToolLabels.CROMWELL);
+        await Ajax(signal).Metrics.captureEvent(Events.applicationCreate, {
+          app: appTools.CROMWELL.label,
+          ...extractWorkspaceDetails(workspace),
+        });
+        await refreshApps(true);
+      } catch (error) {
+        reportError('Cloud Environment Error', error);
+      } finally {
+        setCreating(false);
+      }
+    });
+
+    const pageReady = cbasReady || (currentApp && !getIsAppBusy(currentApp));
+    const launcherDisabled = creating || (currentApp && getIsAppBusy(currentApp)) || (currentApp && !pageReady);
```

Comment:
small nit: I don't have a strong preference, but maybe these 2 can be moved to top where other variables are defined to keep it consistent?